src/main/kotlin/org/rust/ide/intentions/FlattenUseStatementsIntention.kt

```diff
@@ -0,0 +1,167 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsElementTypes
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.RsUseGroup
+import org.rust.lang.core.psi.RsUseSpeck
+import org.rust.lang.core.psi.ext.*
+
+/**
+ * Flatten imports 1 depth
+ *
+ * ```
+ * use a::{
+ *   b::foo,
+ *   b::bar
+ * }
+ * ```
+ *
+ * to this:
+ *
+ * ```
+ * use a::b::foo;
+ * use a::b::bar;
+ * ```
+ */
+class FlattenUseStatementsIntention : RsElementBaseIntentionAction<FlattenUseStatementsIntention.Context>() {
+    override fun getText() = "Flatten use statements"
+    override fun getFamilyName() = text
+
+    interface Context {
+        val useSpecks: List<RsUseSpeck>
+        val root: PsiElement
+        val firstOldElement: PsiElement
+        fun createElements(paths: List<String>, project: Project): List<PsiElement>
+        val oldElements: List<PsiElement>
+        val cursorOffset: Int
+        val basePath: String
+    }
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
```

Comment:
This intention should take into account attributes and visibility:

```rust
#[cfg(foo)]
pub use std::collections::{/*caret*/HashMap, HashSet};
```

```rust
use std::collections::HashMap;
use std::collections::HashSet;
```

Reply:
Should this result in
```rust
#[cfg(foo)]
pub use std::collections::HashMap;
#[cfg(foo)]
pub use std::collections::HashSet;
```
I am specifically asking about the attribute